apps/angular/crud/src/app/error.interceptor.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpInterceptor,
+  HttpEvent,
+  HttpHandler,
+  HttpRequest,
+} from '@angular/common/http';
+import { catchError, finalize, Observable, of } from 'rxjs';
+import { LoaderService } from './loader.service';
+
+@Injectable({ providedIn: 'root' })
+export class ErrorInterceptor implements HttpInterceptor {
+  constructor(private loaderService: LoaderService) {}
+
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    this.loaderService.showLoader();
+    return next.handle(req).pipe(
+      finalize(() => this.loaderService.hideLoader()),
+      catchError((error) => {
+        console.log(error);
+        this.loaderService.hideLoader();
+        return of();
+      })
+    );
+  }
+}
```

Comment:
my personal opinion: this seems great for small application but it will always trigger a loader for http request, and sometimes you don't want to. 

For instance, on this example, we want to trigger a loader per TodoItem when we update it,